Detectors/TRD/workflow/src/TRDEventDisplayFeedSpec.cxx

```diff
@@ -0,0 +1,378 @@
+              << "\t\t\t"
+              << R"("lyr": )" << iLayer << ",\n"
+              << "\t\t\t"
+              << R"("pads": [)"
+              << "\n";
+
+        bool first = true;
+        // for (sauto digit : digits) {
+        for (int iDigit = trigRec.getFirstDigit(); iDigit < trigRec.getFirstDigit() + trigRec.getNumberOfDigits(); ++iDigit) {
+          Digit digit = digits[iDigit];
+
+          int detector = digit.getDetector();
+
+          if (detector == det + iLayer) {
+            // Digits are in stack with track
+            int row = digit.getPadRow();
+            int col = digit.getPadCol();
+
+            if (first) {
+              first = false;
+            } else {
+              *dout << ",\n";
+            }
+
+            *dout << "\t\t\t\t{\n"
+                  << "\t\t\t\t\t"
+                  << R"("row": )" << row << ",\n"
+                  << "\t\t\t\t\t"
+                  << R"("col": )" << col << ",\n"
+                  << "\t\t\t\t\t"
+                  << R"("tbins": [)";
+
+            // Event display draws 24 time bins
+            int nTimebins = 25;
+            int t = 0;
+            for (auto adc : digit.getADC()) {
+              *dout << adc << ((t + 1 < nTimebins) ? ", " : "");
+              t++;
+              if (t == nTimebins) {
+                break;
+              }
+            }
+            *dout << "]\n";
+            *dout << "\t\t\t\t}";
+          }
+        }
+        *dout << "\n\t\t\t]\n";
+        *dout << "\t\t}" << (iLayer + 1 < 6 ? "," : "") << std::endl;
+      }
+      *dout << "\t]\n}";
+
+      dout->flush();
+      dout->close();
+    }
+  }
+}
+
+void TRDEventDisplayFeedSpec::run(o2::framework::ProcessingContext& pc)
+{
+  LOG(info) << "Running event display feed...";
+
+  summary->open(Form("../alice-trd-event-display/data/o2/script.js"));
+  // Path to data file for alicetrd.web: lxplus.cern.ch:/eos/project/a/alice-trd/www/eventdisplay/data/o2/script.js
+
+  *summary << "function getDigitsLoadUrl(eventNo, sector, stack) { return `"
+           << "data/o2/${eventNo}.${sector}.${stack}.json`; }"
+           << std::endl
+           << std::endl;
+
+  *summary << "function getData() {\n\treturn [";
+
+  auto tracks = pc.inputs().get<gsl::span<TrackTRD>>("trdtracks");
+  auto tracklets = pc.inputs().get<gsl::span<Tracklet64>>("trdtracklets");
+  auto digits = pc.inputs().get<gsl::span<Digit>>("trddigits");
+
+  mTrigRecs = pc.inputs().get<gsl::span<TriggerRecord>>("trdtriggerrec");
+  auto trackTrigRecs = pc.inputs().get<gsl::span<TrackTriggerRecord>>("tracktriggerrec");
+
+  // Display only first 5 events
+  int nEvents = 5;
+
+  for (int iEvent = 0; iEvent < nEvents; ++iEvent) {
+    const auto& trigRec = mTrigRecs[iEvent];
+    const auto& trackTrigRec = trackTrigRecs[iEvent];
+
+    if (trackTrigRec.getNumberOfTracks() == 0) {
+      continue;
+    }
+
+    for (int det = 0; det < 540; det++) {
+      mUsedDetectors[det] = false;
+    }
```

Comment:
No need for this if you consider using the std::array with brace-initialization.

Reply:
I think I still need this since I have to reset it to false after each event

Reply:
Still... :), if you change to std::array you can just use here: `mUsedDetectors = { false };`

Reply:
..or you can go for a [std::bitest](https://en.cppreference.com/w/cpp/utility/bitset) :)

Reply:
even cooler :)